lib/inertia_rails/renderer.rb

```diff
@@ -27,13 +27,13 @@ def render
     private
 
     def props
-      only = (@request.headers['X-Inertia-Partial-Data'] || '').split(',').compact.map(&:to_sym)
-
-      _props = ::InertiaRails.shared_data(@controller).merge(@props)
-
-      _props = (only.any? && @request.headers['X-Inertia-Partial-Component'] == component) ?
-        _props.select {|key| key.in? only} :
-        _props
+      _props = ::InertiaRails.shared_data(@controller).merge(@props).select do |key, prop|
+        if rendering_partial_component?
+          key.in? partial_keys
+        else
+          !prop.is_a?(InertiaRails::Lazy)
+        end
+      end
```

Comment:
stuffing some of this code into methods should reveal the intent a bit better

Reply:
boy that was confusing. Good call.